certificates/views.py

```diff
@@ -100,3 +103,48 @@ def get_context_data(self, **kwargs):
         context['certificate'] = certificate
 
         return context
+
+
+class GradeRecordView(TemplateView):
+    """
+    View for grade records
+    """
+
+    template_name = 'grade_record.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        js_settings = {
+            "gaTrackingID": settings.GA_TRACKING_ID,
+            "reactGaDebug": settings.REACT_GA_DEBUG,
+            "edx_base_url": settings.EDXORG_BASE_URL,
+        }
+        context["js_settings_json"] = json.dumps(js_settings)
+
+        enrollment = ProgramEnrollment.objects.get(hash=kwargs.get('record_hash'))
+        user = enrollment.user
+        courses = enrollment.program.course_set.all()
+        mmtrack = get_mmtrack(user, enrollment.program)
+        context["program_title"] = enrollment.program.title
+        context["program_status"] = "completed" if MicromastersProgramCertificate.objects.filter(
+            user=user, program=enrollment.program) else "partially"
```

Comment:
We should use `.exists()` here at the end of the queryset so that the query is specifically for whether the rows exist. https://docs.djangoproject.com/en/2.1/ref/models/querysets/#s-when-querysets-are-evaluated